llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -961,61 +961,45 @@ Value *SPIRVToLLVM::transConvertInst(SPIRVValue *bv, Function *f, BasicBlock *bb
   }
 }
 
-// Decide what fast math flags to set in Builder, just before generating
-// code for BV. Decorations on BV may prevent us from setting some flags.
-FastMathFlags SPIRVToLLVM::getFastMathFlags(SPIRVValue *bv) {
+static FastMathFlags buildFastMathFlags(unsigned flags) {
   FastMathFlags fmf;
 
-  // For floating-point operations, if "FastMath" is enabled, set the "FastMath"
-  // flags on the handled instruction
-  if (const unsigned flags = m_shaderOptions->fastMathFlags) {
-    if (flags & FastMathFlags::AllowReassoc)
-      fmf.setAllowReassoc();
-
-    if (flags & FastMathFlags::NoNaNs)
-      fmf.setNoNaNs();
-
-    if (flags & FastMathFlags::NoInfs)
-      fmf.setNoInfs();
-
-    if (flags & FastMathFlags::NoSignedZeros)
-      fmf.setNoSignedZeros();
+  if (flags & FastMathFlags::AllowReassoc)
+    fmf.setAllowReassoc();
 
-    if (flags & FastMathFlags::AllowReciprocal)
-      fmf.setAllowReciprocal();
-
-    if (flags & FastMathFlags::AllowContract)
-      fmf.setAllowContract();
+  if (flags & FastMathFlags::NoNaNs)
+    fmf.setNoNaNs();
 
-    if (flags & FastMathFlags::ApproxFunc)
-      fmf.setApproxFunc();
+  if (flags & FastMathFlags::NoInfs)
+    fmf.setNoInfs();
 
-    return fmf;
-  }
-  if (SPIRVGenFastMathFlags) {
-    if (SPIRVGenFastMathFlags & FastMathFlags::AllowReassoc)
-      fmf.setAllowReassoc();
+  if (flags & FastMathFlags::NoSignedZeros)
+    fmf.setNoSignedZeros();
 
-    if (SPIRVGenFastMathFlags & FastMathFlags::NoNaNs)
-      fmf.setNoNaNs();
+  if (flags & FastMathFlags::AllowReciprocal)
+    fmf.setAllowReciprocal();
 
-    if (SPIRVGenFastMathFlags & FastMathFlags::NoInfs)
-      fmf.setNoInfs();
+  if (flags & FastMathFlags::AllowContract)
+    fmf.setAllowContract();
 
-    if (SPIRVGenFastMathFlags & FastMathFlags::NoSignedZeros)
-      fmf.setNoSignedZeros();
+  if (flags & FastMathFlags::ApproxFunc)
+    fmf.setApproxFunc();
 
-    if (SPIRVGenFastMathFlags & FastMathFlags::AllowReciprocal)
-      fmf.setAllowReciprocal();
+  return fmf;
+}
 
-    if (SPIRVGenFastMathFlags & FastMathFlags::AllowContract)
-      fmf.setAllowContract();
+// Decide what fast math flags to set in Builder, just before generating
+// code for BV. Decorations on BV may prevent us from setting some flags.
+FastMathFlags SPIRVToLLVM::getFastMathFlags(SPIRVValue *bv) {
+  FastMathFlags fmf;
 
-    if (SPIRVGenFastMathFlags & FastMathFlags::ApproxFunc)
-      fmf.setApproxFunc();
+  // For floating-point operations, if "FastMath" is enabled, set the "FastMath"
+  // flags on the handled instruction
+  if (const unsigned flags = m_shaderOptions->fastMathFlags)
+    return buildFastMathFlags(flags);
```

Comment:
It might be useful to have this option (although I can't think of something right now).
It can't be removed immediately as I have to modify the use of this flag to the disable version in a few places once the change is landed.

I'd prefer that if it does exist, then it is present in the shader options in the .pipe file - otherwise we can get different behaviour in the driver and with a stand alone compilation.